### PR TITLE
Make overlay work again (without the top-level config)

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: examples/vg-specs/overlay_area_short.vg.json [hunk 1]
```diff
@@ -59,22 +59,12 @@
         {
             "name": "layer_1_source",
             "source": "source",
-            "transform": [
-                {
-                    "type": "filter",
-                    "expr": "datum[\"date\"] !== null && !isNaN(datum[\"date\"]) && datum[\"price\"] !== null && !isNaN(datum[\"price\"])"
-                }
-            ]
+            "transform": []
```
Comment (kanitw) [inline examples/vg-specs/overlay_area_short.vg.json]: @domoritz I wonder if something is wrong with filterInvalid for layered spec.

Comment (kanitw) [inline examples/vg-specs/overlay_area_short.vg.json]: but it seems beyond the scope of the PR to fix that.

Comment (kanitw) [inline examples/vg-specs/overlay_area_short.vg.json]: FYI, I added a TODO to https://github.com/vega/vega-lite/issues/2066

Comment (domoritz) [inline examples/vg-specs/overlay_area_short.vg.json]: This looks odd indeed.

File: src/spec.ts [hunk 2]
```diff
@@ -255,60 +254,50 @@ function normalizeRangedUnit(spec: UnitSpec) {
 }
 
 
-// FIXME(#1804): rewrite this
+// FIXME(#1804): re-design this
```
Comment (domoritz) [inline src/spec.ts]: Lol, all this work below and it's not a redesign...

Comment (domoritz) [inline src/spec.ts]: Can you elaborate on what you think is wrong with this. A fixme is a strong comment.

Comment (kanitw) [inline src/spec.ts]: It already refers to https://github.com/vega/vega-lite/issues/1804

**Pull request merged**